### DISCUSSION TITLE
Querying stage selection python

### DISCUSSION THREAD

**Discussion by daviddeacon** [Autodesk/maya-usd#1566] [Q&A] [answered]
Hi,

I'm developing some artist friendly tools to automate working with the USD stage in Maya. I'm used to use `cmds.ls(sl=True)` and `cmds.getAttr(item)` to get information from Maya native nodes, but what would be the equivalent for querying selection and attributes from the stage / prims?

Thanks,
David.

**Comment by kxl-adsk** [comment, marked as answer]
Hi @daviddeacon,

Not sure if you have any preferences in APIs, but here is one way that allows you to get to USD APIs and then list attributes with it.

```import maya.cmds as cmds

from mayaUsd.lib import proxyAccessor as pa
import mayaUsd.lib as mayaUsdLib

from pxr import Usd


shapePath, sdfPath = pa.getSelectedDagAndPrim()
    
stage = mayaUsdLib.GetPrim(shapePath).GetStage()
primPath = Sdf.Path(sdfPath)
prim = stage.GetPrimAtPath(primPath)
print(prim.GetAttributes())
```

Cheers,
KxL

**Reply by daviddeacon** [reply]
Awesome,

This totally got me started.

Thanks,
David.
